### PR TITLE
[Payments NOX] Implement different post sandbox account setup modal for dev mode

### LINKED ISSUE #53668: [WC Payments NOX] Dev/builder mode post test-drive account creation modal
### Description
We need to implement the post test-drive account creation modal according to the design:

Image

### Dev notes

Reactifying the Payments settings page: pe4Hk0-1ly-p2
Populate react-powered main payments settings screen PR: https://github.com/woocommerce/woocommerce/pull/50825

### Designs
Design post: pfHfG4-q7-p2
Figma: xO8YdpKfaVoP2d8NvweZ5I-fi-1428_21437

### Additional context
PRD: 1wMfX8CAjDnOp24tcU-PADB0pOt3cz7mGdyZYeRBxx2Y-gdoc

### PR DESCRIPTION
**Pull request by oaratovskyi**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Implement different post sandbox account setup modal for dev mode

Closes #53668 .

### How to test the changes in this Pull Request:

1. Create a new JurassicNinja site with WooCommerce on this PR's live branch and the WooCommerce Beta Tester plugin (here is a [direct JN create link](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce-beta-tester&woocommerce&branches.woocommerce=dev/53668-post-test-drive-setup-modal))
2. Install WCPay Dev Tools plugin
3. Go to WP admin > Tools > WCA Test Helper > Features and make sure the `reactify-classic-payments-settings` is enabled
4. Access the WP admin of the newly created site, click on the WooCommerce menu item, and you should be taken to the WooCommerce profiler. Click "Skip guided setup" and choose `US` as the store's base location country.
5. Go to the Payments -> Settings page, Install and Enable WooPayments.
6. Wait for Jetpack popup, approve it and you'll get navigated to test drive onboarding flow.
7. After it's done, check that you see this modal

image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/client/admin/client/settings-payments/settings-payments-main.tsx [hunk 1]
```diff
@@ -42,8 +43,10 @@ export const SettingsPaymentsMain = () => {
 		PAYMENT_SETTINGS_STORE_NAME
 	);
 	const [ errorMessage, setErrorMessage ] = useState< string | null >( null );
-	const [ livePaymentsModalVisible, setLivePaymentsModalVisible ] =
-		useState( false );
+	const [
+		postSandboxAccountSetupModalVisible,
```
Comment (anu-rock) [inline plugins/woocommerce/client/admin/client/settings-payments/settings-payments-main.tsx:47]: It's a bit too verbose, I think. WDYT about `sandboxSuccessModalVisible`?

Comment (anu-rock) [reply plugins/woocommerce/client/admin/client/settings-payments/settings-payments-main.tsx:47]: But care less about this suggestion. It's a super-tiny nitpick.

File: plugins/woocommerce/client/admin/client/settings-payments/utils.ts [hunk 2]
```diff
@@ -63,3 +63,15 @@ export const providersContainWooPaymentsInTestMode = (
 		!! wooPayments?.state?.test_mode && ! wooPayments?.state?.needs_setup
 	);
 };
+
+/**
+ * Checks whether providers contain WooPayments gateway in dev mode that is set up.
+ *
+ * @param providers payment providers
+ */
+export const providersContainWooPaymentsInDevMode = (
```
Comment (anu-rock) [inline plugins/woocommerce/client/admin/client/settings-payments/utils.ts:72]: This name is definitely too jargony. I suggest simply `isWooPaymentsInDevMode`.

**Review by anu-rock — APPROVED** [review]
✅  Code looks good. (left very minor comments)
✅  Tested as expected.

File: plugins/woocommerce/client/admin/client/settings-payments/settings-payments-main.tsx (hunk 1 above)
Comment (oaratovskyi) [reply plugins/woocommerce/client/admin/client/settings-payments/settings-payments-main.tsx:47]: okay, I'll leave it as is

File: plugins/woocommerce/client/admin/client/settings-payments/utils.ts (hunk 2 above)
Comment (oaratovskyi) [reply plugins/woocommerce/client/admin/client/settings-payments/utils.ts:72]: maybe in future PR! it's the same with the sandbox mode

**Pull request merged**